api/src/services/__tests__/article-tests.ts

```diff
@@ -144,20 +145,61 @@ describe('Article', () => {
             const savedArticle = await response.json<IArticle>();
             chai.expect(savedArticle.publishedAt).to.equal(article.publishedAt);
         }));
+
+        describe('When getting sections', () => {
+            it('should returns the article\'s ID and its sections', mochaAsync(async () => {
+                const sections = {
+                    List: [
+                        { name: 'Marketing' },
+                        { id: 1, name: 'Tutorial' }
+                    ]
+                };
+                const response = await fetchWithLogin('http://localhost:1337/api/article/2/sections', {
+                    method: 'POST',
+                    body: JSON.stringify(sections),
+                    headers: {
+                        'Content-Type': 'application/json'
+                    }
+                });
+                const returnedObject = await response.json<{articleId: number, sections: ISection[]}>();
+                chai.expect(returnedObject.articleId).to.equal(2);
+                chai.expect(returnedObject.sections[1]).to.deep.equal(sections.List[1]);
+                chai.expect(returnedObject.sections.length).to.equal(2);
+            }));
+        });
+        describe('When giving an non existing section', () => {
+            it('should returns a sql error', mochaAsync(async () => {
+                const sections = {
+                    List: [
+                        { name: 'Marketing' },
+                        { id: 15, name: 'Tutorial' }
+                    ]
+                };
+                const response = await fetchWithLogin('http://localhost:1337/api/article/2/sections', {
+                    method: 'POST',
+                    body: JSON.stringify(sections),
+                    headers: {
+                        'Content-Type': 'application/json'
+                    }
+                });
+                const returnedObject = await response.json<{error: string}>();
+                chai.expect(returnedObject.error).to.equal('Bad request : SequelizeForeignKeyConstraintError: SQLITE_CONSTRAINT: FOREIGN KEY constraint failed');
+            }));
+        });
```

Comment:
You're missing tests for section creation and deletion.

You should add connected-only `GET /sections` and `GET /article-sections` to use is your tests to check if the database was properly updated.

Nothing fancy, just a `findAll` call being a `req.user.signedIn` test. Don't make a swagger definition, they're there only for testing.